woocommerce/woocommerce-ios
# [Mobile Payments] Fix trying out Tap to Pay is no-op from the About TTP screen

Closes: #14078 


## Why

In Payments > About Tap To Pay > Set Up Tap To Pay on iPhone > Try a Payment: in production, nothing happens after the spinner stops as in the "before" screencast below. This is confusing and a broken flow as the merchant cannot test out the payment with Tap To Pay.

## How

From debugging, the issue was because `navigationDestination` does not work when the SwiftUI view is not within a `NavigationStack`/`NavigationView`. In this PR, the top-level view is now wrapped in a `NavigationStack` and the navigation works as expected.

## Steps to reproduce


1. Delete the app from your phone to clear the saved TTP transaction state
2. Log in to a TTP-eligible site
3. Navigate to Menu > Payments > About Tap to Pay on iPhone
4. Tap Set up Tap to Pay on iPhone
5. Go through the set up process
6. Tap Try a Payment --> it should navigate to the test payment page
7. Proceed with the test payment --> navigation should work as expected

## Testing information


- [x] @jaclync perform a confidence check on another entry point to TTP:

1. Delete the app from your phone to clear the saved TTP transaction state
2. Log in to a TTP-eligible site
3. Navigate to Menu > Payments > Set Up Tap to Pay on iPhone
4. Tap Set up Tap to Pay on iPhone
5. Go through the set up process
6. Tap Try a Payment --> it should navigate to the test payment page
7. Proceed with the test payment --> navigation should work as expected

- [x] @jaclync perform a confidence check on onboarding in TTP

## Screenshots


before:

https://github.com/user-attachments/assets/514b8890-a868-4678-98ff-a8d1ca673029

after:

https://github.com/user-attachments/assets/db7d1c09-10fa-435e-a401-3afa364699a9



---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
